CMakeLists.txt

```diff
@@ -13,19 +15,20 @@ option(MAGIC_ENUM_OPT_BUILD_EXAMPLES "Build magic_enum examples" ${IS_TOPLEVEL_P
 option(MAGIC_ENUM_OPT_BUILD_TESTS "Build and perform magic_enum tests" ${IS_TOPLEVEL_PROJECT})
 option(MAGIC_ENUM_OPT_INSTALL "Generate and install magic_enum target" ${IS_TOPLEVEL_PROJECT})
 
+enable_testing()
 if(MAGIC_ENUM_OPT_BUILD_EXAMPLES)
     add_subdirectory(example)
 endif()
 
 if(MAGIC_ENUM_OPT_BUILD_TESTS)
-    enable_testing()
     add_subdirectory(test)
 endif()
 
 include(CMakePackageConfigHelpers)
 
 add_library(${PROJECT_NAME} INTERFACE)
 add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+target_compile_features(${PROJECT_NAME} INTERFACE cxx_std_${CMAKE_CXX_STANDARD})
```

Comment:
I remember one problem related to this https://github.com/Neargye/magic_enum/issues/42, but I do not know if it is now

Reply:
This is some special. MSVC uses the latest, mapped correctly in CMake if configured correctly. So the CMAKE_CXX_STANDARD flag should know the version is correct.

But what can we do to build some test cases for that to see whether this is an issue or not?